x-pack/plugins/security_solution/server/lib/telemetry/receiver.ts

```diff
@@ -1320,4 +1347,225 @@ export class TelemetryReceiver implements ITelemetryReceiver {
+    };
+
+    return es.indices
+      .getDataStream(request)
+      .then((response) =>
+        response.data_streams.map((ds) => {
+          return {
+            datastream_name: ds.name,
+            indices:
+              ds.indices?.map((index) => {
+                return {
+                  index_name: index.index_name,
+                  ilm_policy: index.ilm_policy,
+                } as Index;
+              }) ?? [],
+          } as DataStream;
+        })
+      )
+      .catch((error) => {
+        this.logger.warn('Error fetching datastreams', { error_message: error } as LogMeta);
+        throw error;
+      });
+  }
+
+  public async *getIndicesStats(indices: string[], config: QueryConfig) {
+    const es = this.esClient();
+
+    this.logger.debug(`Fetching indices stats for ${indices.length} indices`, {
+      indices,
+    } as LogMeta);
+
+    const groupedIndices = findCommonPrefixes(indices, config).map((v, _) => v.parts);
+
+    this.logger.debug(`Splitted indices into ${groupedIndices.length} groups`, {
+      groups: groupedIndices.length,
+    } as LogMeta);
+
+    for (const group of groupedIndices) {
+      for (const name of group) {
+        // this.logger.debug(`Fetching indices stats for ${name}`, { name } as LogMeta);
+        const request: IndicesStatsRequest = {
+          index: `${name}*`,
+          level: 'indices',
+          metric: ['docs', 'search', 'store'],
+          expand_wildcards: ['open', 'hidden'],
+          filter_path: [
+            'indices.*.total.search.query_total',
+            'indices.*.total.search.query_time_in_millis',
+            'indices.*.total.docs.count',
+            'indices.*.total.docs.deleted',
+            'indices.*.total.store.size_in_bytes',
+          ],
+        };
+
+        try {
+          const response = await es.indices.stats(request);
+          for (const [indexName, stats] of Object.entries(response.indices ?? {})) {
+            yield {
+              index_name: indexName,
+              query_total: stats.total?.search?.query_total,
+              query_time_in_millis: stats.total?.search?.query_time_in_millis,
+              docs_count: stats.total?.docs?.count,
+              docs_deleted: stats.total?.docs?.deleted,
+              docs_total_size_in_bytes: stats.total?.store?.size_in_bytes,
+            } as IndexStats;
+          }
+        } catch (error) {
+          this.logger.warn('Error fetching indices stats', { error_message: error } as LogMeta);
+          throw error;
+        }
+      }
+    }
+  }
+
+  public async *getIlmsStats(indices: string[], config: QueryConfig) {
+    const es = this.esClient();
+
+    this.logger.debug(`Fetching ilm stats for ${indices.length} indices`, { indices } as LogMeta);
+
+    const groupedIndices = findCommonPrefixes(indices, config).map((v, _) => v.parts);
+
+    this.logger.debug(`Splitted indices into ${groupedIndices.length} groups`, {
```

Comment:
Perhaps it's better to use `logger.l` for these kinds of logs so we can troubleshoot them later on